lib/Dialect/FIRRTL/FIRRTLUtils.cpp

```diff
@@ -126,6 +126,71 @@ void circt::firrtl::emitConnect(ImplicitLocOpBuilder &builder, Value dst,
   builder.create<StrictConnectOp>(dst, src);
 }
 
+// This is usable any place an implicit extension might happen.  It will force
+// a disconnect of width inference corrosponding to allowing a lhs < rhs.
+Value firrtl::expandPassiveType(ImplicitLocOpBuilder &builder,
+                                FIRRTLBaseType dstType, Value rhs) {
```

Comment:
Consider using `FIRRTLValue` (alias for `TypedValue<FIRRTLType>`) to avoid the need for the cast and be more explicit to callers about the requirement.